### ISSUE TITLE
Question about using this vs a naive axis-by-axis sweep

### ISSUE THREAD

**Issue by MCArth** [fenomas/voxel-aabb-sweep#2]
Hi, when making the changes for https://github.com/fenomas/voxel-physics-engine/pull/4 I found myself wondering the practical effect of using this instead of a naive axis-by-axis sweep. Did you implement this as a theoretical idea or did you find in practice that the axis-by-axis sweep felt "off"/wonky when moving?

I ask as I need to (re-)implement preventing a player from falling off the edge of blocks and while the exact logic will be different, it seems to me similar tradeoffs apply (complexity vs "feel")

**Comment by fenomas** [comment]
Hi, the reason for all the complexity here is so that the module can correctly handle long-distance sweeps. E.g. if you sweep an AABB 15 blocks in a diagonal direction, you'd get the wrong answer if you decomposed that into two horizontal sweeps.

**Comment by MCArth** [comment]
Ah I see - out of interest, are you using this in your own game? Trying to understand the use case for long distance sweeps. I haven't come across this before (haven't done much physics programming, as you can probably tell. maybe there's a blog that discusses this?)

**Comment by fenomas** [comment]
The physics engine itself would only do long sweeps in the case of something moving very fast, though in principle it could be broken up into multiple integrations. But I do use this module for long distances in other places.

E.g. in noa when you have the camera zoomed out, it checks for obstructions so that the camera doesn't go inside geometry right? Originally I did that with a raycast, but I had a really hard time with edge cases where the camera would wind up too close to some terrain (meaning you'd see through the world because of frustum culling). For a while I tried raycasting and then rolling back a bit, but that felt weird because the amount of rollback would depend on the angle from which the raycast hit the terrain, etc. But then I finally realized that I could just position an small AABB on the camera target, and sweep that back the camera zoom distance, and then place the camera wherever the AABB wound up, and be sure that it would always be a fixed distance away from the nearest solid voxel.

Another place I use it is for a "blink" mechanic in my game, where the player is meant to be teleporting forwards a fixed distance - I just sweep their AABB by the desired distance, so that I don't have to worry about cases where they'd blink through a geometry, etc. Does that help?
